bundles/org.openhab.binding.deconz/src/main/java/org/openhab/binding/deconz/internal/handler/LightThingHandler.java

```diff
@@ -70,21 +77,51 @@
 
     private final Logger logger = LoggerFactory.getLogger(LightThingHandler.class);
 
+    private final StateDescriptionProvider stateDescriptionProvider;
+
     private long lastCommandExpireTimestamp = 0;
+    private boolean needsPropertyUpdate = false;
 
     /**
      * The light state. Contains all possible fields for all supported lights
      */
     private LightState lightStateCache = new LightState();
     private LightState lastCommand = new LightState();
 
-    private final int ct_max;
-    private final int ct_min;
+    // set defaults, we can override them later if we receive better values
+    private int ctMax = ZCL_CT_MAX;
+    private int ctMin = ZCL_CT_MIN;
 
-    public LightThingHandler(Thing thing, Gson gson) {
+    public LightThingHandler(Thing thing, Gson gson, StateDescriptionProvider stateDescriptionProvider) {
         super(thing, gson);
-        ct_max = parseIntWithFallback(thing.getProperties().get(PROPERTY_CT_MAX), ZCL_CT_MAX);
-        ct_min = parseIntWithFallback(thing.getProperties().get(PROPERTY_CT_MIN), ZCL_CT_MIN);
+
+        this.stateDescriptionProvider = stateDescriptionProvider;
+    }
+
+    @Override
+    public void initialize() {
+        if (thing.getThingTypeUID().equals(THING_TYPE_COLOR_TEMPERATURE_LIGHT)
+                || thing.getThingTypeUID().equals(THING_TYPE_EXTENDED_COLOR_LIGHT)) {
```

Comment:
Would there ever be a circumstance in which a thing would contain `CT_MIN` and `CT_MAX` properties without the thingtype being `THING_TYPE_COLOR_TEMPERATURE_LIGHT` or `THING_TYPE_EXTENDED_COLOR_LIGHT`. Would it be possible to do away with the thingtype check here and only check for the presence of the properties instead?

Reply:
Unfortunately not. If you add the thing manually (either via PaperUI or .things-file) the properties are not present. We need to add them after we received these values. For unmanaged things this needs to be checked every time the thing is constructed.

For auto-discovered things this is not necessary as the properties are added during discovery.